lib/core/src/Cardano/Wallet.hs

```diff
@@ -1819,6 +1788,30 @@ balanceTransactionWithSelectionStrategy
             (\(_,o) -> (o, Nothing))
                 <$> L.find (\(i',_) -> i == i') (extraInputs update)
 
+
+    -- Increasing the fee by much should only happen if coin-selection is unable
+    -- to construct a change output respecting the minUTxOValue.
+    guardReasonableExcessFee
+        :: TxFeeAndChange
+        -> TxFeeAndChange
+    guardReasonableExcessFee fc@(TxFeeAndChange feeToBurn _)
+        | feeToBurn > minAda <> minAda =
```

Comment:
I think technically this could be problematic on a network where the fee increase derived from coin-value-encoding is greater than 2*minAda — e.g. if minAda is tiny or 0...

Maybe we can add `feePerByte * 16` to the right side… (or keep the static 20 ada one 🤷‍♂️ if it gets too complicated)

Reply:
I ended up just dropping `guardReasonableExcessFee` completely. I think it's more trouble than it's worth.